datafusion/datasource-csv/src/file_format.rs

```diff
@@ -642,6 +643,38 @@ impl BatchSerializer for CsvSerializer {
         drop(writer);
         Ok(Bytes::from(buffer))
     }
+
+    async fn deserialize(
```

Comment:
I wonder how this is related to https://docs.rs/datafusion/latest/datafusion/physical_plan/insert/trait.DataSink.html

Specifically, DataSink already has some way to serialize batches -- do we need another one?

What is the usecase of this code as it isn't used to implement any built in DataFusion features? Maybe we could add an example or something that shows how it works